### PR TITLE
Add initial Device resource  documentation.

### PR DESCRIPTION
**Pull request by davischristi**
Adding initial Device resource documentation.

### CODE REVIEW

File: lib/resources/dstu2/device.yaml [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+            ],
+          "text": "Serial Number"
+        },
+        "value": "SERIALNBRDA88ABA9-9FAD-4A03-9E9A-B52299AE9AE4"
+        }
+      ]
+    }
+  children:
+
+  - name: type
```
Comment (daliboz) [inline lib/resources/dstu2/device.yaml]: This isn't linking to anything right now - I don't think this is intended to be a child of identifier, right?

File: lib/resources/dstu2/device.yaml [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+    {
+      "type": {
+        "text": "Cochlear Implant Right Ear"
+      }
+    }
+  binding:
+    description: The type of device.
+    note: Currently not bound to any terminology and only 'Device.type.text' will be returned as of now.
+
+- name: manufacturer
```
Comment (daliboz) [inline lib/resources/dstu2/device.yaml]: Is this or below actually being used anywhere?

**Comment by daliboz** [comment]
Should we add a note about not supporting the status modifier element?

File: lib/resources/dstu2/device.yaml (hunk 2 above)
Comment (cerbs) [inline lib/resources/dstu2/device.yaml]: Nope. They aren't used. I'll yank 'em

**Comment by cerbs** [comment]
Added status modifier note

**Comment by cerbs** [comment]
👍  After updates

**Comment by cerbs** [comment]
Rebasing onto master

**Comment by cerbs** [comment]
Merged and squashed on 224dc4c

**Pull request closed without merging**